workflow/torch/requires_grad.py

```diff
@@ -9,20 +9,17 @@ def __init__(self, module, grad=True):
 
         self.module = module
         self.grad = grad
-        self.requires_grads = [
-            p.requires_grad for p in module.parameters()
-        ]
+        self.named_requires_grads = { name: parameter.requires_grad for (name, parameter) in module.named_parameters() }
```

Comment:
Dict and zip instead?

Reply:
sure, tried it, I thought it was going to be worse then it was

Reply:
I didn't realize we needed to access the attribute `requires_grad`. I prefer the old way actually but this is fine

Reply:
"old way" as in dict comprehension?

Reply:
Yes :)